src/features/impl_rkyv.rs

```diff
@@ -0,0 +1,192 @@
+#[cfg(feature = "bytecheck")]
+macro_rules! impl_rkyv {
+    (@bytecheck $type:ty) => {
+        impl<C: ?Sized> bytecheck::CheckBytes<C> for $type {
+            type Error = core::convert::Infallible;
+
+            #[inline]
+            unsafe fn check_bytes<'a>(
+                value: *const Self,
+                _: &mut C,
+            ) -> Result<&'a Self, Self::Error> {
+                Ok(&*value)
+            }
+        }
+    };
+
+    ($type:ty) => {
+        impl_rkyv_derive!(@serialize $type);
+        impl_rkyv_derive!(@archive_deserialize $type);
+        impl_rkyv!(@bytecheck $type);
+    };
+}
+
+#[cfg(not(feature = "bytecheck"))]
+macro_rules! impl_rkyv {
+    ($type:ty) => {
+        impl_rkyv_derive!(@serialize $type);
+        impl_rkyv_derive!(@archive_deserialize $type);
+    };
+}
+
+macro_rules! impl_rkyv_derive {
+    (@serialize $type:ty) => {
+        impl<S: Fallible + ?Sized> Serialize<S> for $type {
+            #[inline]
+            fn serialize(&self, _: &mut S) -> Result<Self::Resolver, S::Error> {
+                Ok(())
+            }
+        }
+    };
+
+    (@archive_deserialize $type:ty) => {
+        const _: () = {
```

Comment:
Why is this block necessary?

Reply:
Looks like scoping for the type alias right below (`type Archived = $type`). Doesn't look strictly necessary to me.

Reply:
Yes, it was to scope the type alias. I removed the scope.